extension/scripts/Selector/SelectorTable.js

```diff
@@ -106,42 +119,52 @@ var SelectorTable = {
         return result;
     },
 
+    isEmptyObject: function (obj) {
+    for (var i in obj) {
+        if (obj.hasOwnProperty(i)) {
+            return false;
+        }
+    }
+    return true;
+    },
     _getData: function (parentElement) {
 
         var dfd = $.Deferred();
 
         var tables = this.getDataElements(parentElement);
-
+        var verticalTable = this.verticalTable;
+        if (this.verticalTable === undefined) {
+            verticalTable = $("#edit-selector [name=verticalTable]").is(":checked");
+        }
         var result = [];
         $(tables).each(function (k, table) {
-            var headerCells = this.getTableHeaderColumns($(table)),
-                dataSelector = this.getTableDataRowSelector(),
-                objKeys = Object.keys(headerCells),
-                headerCellCount = objKeys.length,
-                isVerticalHeader = headerCellCount && headerCells[Object.keys(headerCells)[0]].isVerticalHeader;
+            var headerCells = this.getTableHeaderColumns($(table));
+            var dataSelector = this.getTableDataRowSelector();
+
+
 
-            if (isVerticalHeader) {
+            if (verticalTable) {
                 var results = this.getVerticalDataCells(table, dataSelector);
-                result.push.apply(result, results);
-            } else {
+                for (i = 0;i<results.length;i++) {
+                    if (!this.isEmptyObject(results[i])) result.push(results[i]);
+                }
+                } else {
                 $(table).find(dataSelector).each(function (i, dataCell) {
                     var data = {};
 
                     this.columns.forEach(function (headerCell) {
-                        if (headerCell.extract === true) {
-                            if (headerCells[headerCell.header] === undefined) {
-                                data[headerCell.name] = null;
-                            }
-                            else {
-                                var header = headerCells[headerCell.header];
+                                var header = headerCells[headerCell.header.replace(/^\s*/,'').replace(/\s*$/,'')];
                                 var rowText = $(dataCell).find(">:nth-child(" + header.index + ")").text().trim();
-                                data[headerCell.name] = rowText;
-                            }
-                        }
+                                if (headerCell.extract){
+                                    data[headerCell.name.replace(/^\s*/,'').replace(/\s*$/,'')] = rowText;
```

Comment:
Why you need 2 replaces here? Seems that you should already have to this moment processed result keys.

Reply:
I found this method of removing spaces on a forum with a lot of likes. I consider it necessary to use the working version, and not to invent a Bicycle that may not go.